### ISSUE TITLE
Emoji icons are cut off in outlined mode with preset height

### ISSUE THREAD

**Issue by johnaziz57** [callstack/react-native-paper#2495]
### Current behaviour

When:
* Setting a height for the TextInput
* Start typing
* Adding emojis to the input

Result: 
* Emojis are cut off from the top 

### Expected behaviour

* Emojis not to be cut off when height is set

### Code sample
[Expo](https://snack.expo.io/@john_aziz57/react-native-paper-emoji-bug)

```
import * as React from 'react';
import { View } from 'react-native';
import { TextInput } from "react-native-paper"

export default function App() {
  return (
    <View style={{
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#ecf0f1',
    padding: 8,
  }}>
    <TextInput 
    mode="outlined"
    style={{
      backgroundColor: "lightblue",
      height: 48
    }}/>
    </View>
  );
}

```

### Screenshots (if applicable)

![Screenshot_20210111-210039](https://user-images.githubusercontent.com/14923445/104232187-4663e200-5450-11eb-855c-099371aed5a6.jpg)

### What have you tried

* Changing the height to be bigger or smaller -> same faulty result
* Changing the padding, TopPadding, VerticalPadding -> same faulty result
* Wrapping the TextInput with another View with the desired height -> TextInput didn't respect the height of the parent
* Using lineHeight instead of height -> TextInput ignores the line height and draws the whole view bigger than intended
* Using lineHeight along with height -> same faulty result

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | iOS
| react-native          | 0.63.3
| react-native-paper    | 4.6.0
| node                  | 12.16.1
| npm or yarn           | yarn 1.22.4
| expo sdk              | 40.0.0
| react-native-vector-icons| ^7.0.0

Any further info required, I will be glad to provide

# Update 1 (12-01-21, 11:40):
After reading the code, I see that the padding is calculated and no padding values whether it was `paddingTop` or `paddingBottom` are not take into consideration

# Update 2 (12-01-21, 12:12):
I managed to solve the problem if only it is allowed to override the calculated padding
I did something similar to this in `TextInputOutlined.tsx:176`, if this sounds like a sound idea I would be happy to open a PR
```
    const pad = calculatePadding(paddingSettings);
    let paddingOut = adjustPaddingOut({ ...paddingSettings, pad });

    if (paddingVertical) {
      if (typeof paddingVertical !== 'number') {
        console.warn('Currently we support only numbers in paddingVertical prop');
      } else {
        paddingOut = {
          paddingTop: paddingVertical,
          paddingBottom: paddingVertical
        }
      }
    }
```

**Comment by johnaziz57** [comment]
Tried with the latest `react-native` and `react-native-vector-icons` still get the same behaviour

**Comment by maltoze** [comment]
Try using `lineHeight` instead of `height`.

**Comment by johnaziz57** [comment]
> Try using `lineHeight` instead of `height`.

I tried but it seems that TextInput is drawn bigger than intended

here I used `lineHeight: 48`, and it is rendered as `56`
![Screenshot_20210111-210039](https://user-images.githubusercontent.com/14923445/104287804-421edf80-54b7-11eb-9e96-cafb768e083d.jpg)

Using both height and lineHeight results in the same cut off

**Comment by magrinj** [comment]
I've fix this issue in #2621, when it's merge you'll be able to provide the `lineHeight` in the style to prevent this cut
